mcp/cmd_test.go

```diff
@@ -239,6 +239,73 @@ func createServerCommand(t *testing.T, serverName string) *exec.Cmd {
 	return cmd
 }
 
+func TestCommandTransportTerminateDuration(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("requires POSIX signals")
+	}
+	requireExec(t)
+
+	tests := []struct {
+		name            string
+		duration        time.Duration
+		wantMaxDuration time.Duration
+	}{
+		{
+			name:            "default duration (zero)",
+			duration:        0,
+			wantMaxDuration: 6 * time.Second, // default 5s + buffer
+		},
+		{
+			name:            "below minimum duration",
+			duration:        500 * time.Millisecond,
+			wantMaxDuration: 6 * time.Second, // should use default 5s + buffer
+		},
+		{
+			name:            "custom valid duration",
+			duration:        2 * time.Second,
+			wantMaxDuration: 3 * time.Second, // custom 2s + buffer
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			// Use a command that won't exit when stdin is closed
+			cmd := exec.Command("sleep", "3600")
```

Comment:
If something goes wrong, this test could wait an hour (in theory; I believe tests time out after 10 min by default). What about 20 seconds?